src/main/java/org/opensearch/knn/index/KNNIndexShard.java

```diff
@@ -171,6 +173,11 @@ List<EngineFileContext> getEngineFileContexts(IndexReader indexReader, KNNEngine
                     String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.L2.getValue());
                     SpaceType spaceType = SpaceType.getSpace(spaceTypeName);
                     String modelId = fieldInfo.attributes().getOrDefault(MODEL_ID, null);
+                    String indexDescription = null;
+                    if (KNNEngine.FAISS.equals(knnEngine)
```

Comment:
nit: Is this `if` branching required? let it return null if it doesn't find it in the fieldinfo. You have to handle null case irrespective of the branching